payload/context/register_test.go

```diff
@@ -19,53 +19,62 @@ import (
 
 type registerSuite struct {
 	testing.IsolationSuite
+
+	hookCtx *stubRegisterContext
+	command RegisterCmd
 }
 
 var _ = gc.Suite(&registerSuite{})
 
-func (registerSuite) TestInitNilArgs(c *gc.C) {
-	r := RegisterCmd{}
-	err := r.Init(nil)
+func (s *registerSuite) SetUpTest(c *gc.C) {
+	s.IsolationSuite.SetUpTest(c)
+
+	s.hookCtx = &stubRegisterContext{}
+
+	s.command = RegisterCmd{
+		hookContextFunc: func() (Component, error) {
+			return s.hookCtx, nil
+		},
+	}
+}
+
+func (s registerSuite) TestInitNilArgs(c *gc.C) {
+	err := s.command.Init(nil)
```

Comment:
All test functions need to be '*' in their receivere.
(s *registerSuite)

Otherwise it does bad things with base classes like IsolationSuite (we track things in AddCleanup that do bad things if you use a value rather than a pointer)

There are several places where this needs to be fixed.

Are there any changes here that aren't just pulling 's.command' into SetUp() (and changing hook context into an acquisition function)? I didn't see any.